bson/src/main/org/bson/codecs/pojo/PojoBuilderHelper.java

```diff
@@ -118,31 +171,47 @@
         }
     }
 
-    private static <T> FieldModelBuilder<T> getFieldBuilder(final Field field, final Class<T> clazz) {
-        return FieldModel.<T>builder(field);
+    private static TypeData<?> getTypeData(final Method method) {
+        if (isGetter(method.getName())) {
+            return getTypeData(method.getGenericReturnType(), method.getReturnType());
+        } else {
+            if (method.getGenericParameterTypes().length != 1) {
```

Comment:
Similarly, these should not be considered get/set methods
  
     int getfoo();  
     void setfoo(int foo);
     int get();
     void set(int foo);

In other words there has to be at least one character after get/set prefix and it has to be a capital letter (using Character.isUpper(char c)).